### PR TITLE
Warn about unused whitelist options

### PR DESCRIPTION
**Pull request by sanxiyn**
I welcome suggestions on better ways to report warnings.

cc #1055

### CODE REVIEW

File: src/regex_set.rs [hunk 1]
```diff
@@ -29,12 +32,23 @@ impl RegexSet {
         &self.items[..]
     }
 
+    pub fn get_unmatched_items(&self) -> Vec<String> {
```
Comment (emilio) [inline src/regex_set.rs]: nit: I'd call this just `unmatched_items`, and travis is complaining because this needs docs :)

**Comment by sanxiyn** [comment]
Now I measured performance on internal codebase and I couldn't detect any change, but maybe our use cases are too small. Benchmark suggestions are welcome.

```
Use case #1

--whitelist-type 1
--whitelist-var 3

old
  time:   770.848 ms.   parse
  time:   632.237 ms.   compute_whitelisted_and_codegen_items
new
  time:   762.946 ms.   parse
  time:   629.872 ms.   compute_whitelisted_and_codegen_items
```

```
Use case #2

--whitelist-function 6
--whitelist-type 5
--whitelist-var 1

old
  time:   878.813 ms.   parse
  time:   743.636 ms.   compute_whitelisted_and_codegen_items
new
  time:   881.604 ms.   parse
  time:   726.318 ms.   compute_whitelisted_and_codegen_items
```

**Comment by emilio** [comment]
How many regexes are you using in your codebase? In Gecko we use quite a lot of them, see all the lists in:

  https://searchfox.org/mozilla-central/source/layout/style/ServoBindings.toml

I still think a configuration option for this would be useful, because it wouldn't be great to spam other developers about warnings that they don't know how to fix, or platform-dependent stuff.

It'd be great if you could do that but otherwise let me know and I'll try to take some time to make it happen as a followup to this PR before the next release.

**Comment by sanxiyn** [comment]
I already gave how many regexes we use. My understanding of RegexSet implementation is that slowdown happens only if multiple regexes actually match, and that doesn't seem to happen in Gecko, and I expect such cases to be rare.

I think it is very desirable to have this on by default, like Rust's unused warnings. I expect it won't be used in practice if it is not on by default. If it isn't too much trouble, can you run the benchmark for Gecko's use case?

I am uninterested in putting this behind a flag, so I appreciate your offer of a followup. Thanks.

**Comment by emilio** [comment]
Alright, this works for now, sorry for the lag (Christmas).

Will followup with the option, thank you for the patch! I agree this is useful :)

**Pull request merged**